core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -57,6 +58,13 @@ final class HttpClientDelegate implements HttpClient {
 
     @Override
     public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
+        final Throwable throwable = ClientPendingThrowableUtil.pendingThrowable(ctx);
+        if (throwable != null) {
+            final UnprocessedRequestException cause = UnprocessedRequestException.of(throwable);
+            handleEarlyRequestException(ctx, req, cause);
+            return HttpResponse.ofFailure(cause);
```

Comment:
Optional; Could we dedupe the code block by adding a method like:
```java
return handleFailedRequest(ctx, req, throwable)?
```

Reply:
Named as `earlyFailedResponse` because `handleFailedRequest` seemed indistinguishable from `handleEarlyRequestException` 😄

Reply:
👍 Sounds better.